Ginger/GingerCore/Drivers/JavaDriverLib/JavaDriver.cs

```diff
@@ -2965,56 +2971,184 @@ private void GetPOMAction(PayLoad payLoad)
             OnRecordingEvent(args);
         }
 
-        private ElementInfo GetElementInfoFromActionConfiguration(ElementActionCongifuration elementActionCongifuration)
+        private ElementInfo GetElementInfoFromActionConfiguration(ElementActionCongifuration elementActionCongifuration, ElementInfo elementInfo)
         {
-            JavaElementInfo elementInfo = new JavaElementInfo();
             try
             {
-                elementInfo.XPath = Convert.ToString(elementActionCongifuration.LocateValue);
-                elementInfo.ElementType = nameof(JavaElementInfo);
-                elementInfo = (JavaElementInfo)((IWindowExplorer)this).LearnElementInfoDetails(elementInfo);
-
                 elementInfo.ElementName = elementActionCongifuration.Description;
+
+                elementInfo = ((IWindowExplorer)this).LearnElementInfoDetails(elementInfo);
+
                 if (Enum.IsDefined(typeof(eElementType), Convert.ToString(elementActionCongifuration.Type)))
                 {
                     elementInfo.ElementTypeEnum = (eElementType)Enum.Parse(typeof(eElementType), Convert.ToString(elementActionCongifuration.Type));
                 }
 
+                if (isPOMRecording && elementInfo.GetType().Equals(typeof(HTMLElementInfo)))
+                {
+                    elementInfo.Properties.Add(new ControlProperty() { Name = ElementProperty.ParentBrowserPath, Value = parentBrowserPath });
+                }
+
                 foreach (var elementLocator in elementInfo.Locators)
                 {
                     elementLocator.Active = true;
                     elementLocator.IsAutoLearned = true;
                 }
-
             }
             catch (Exception ex)
             {
                 Reporter.ToLog(eLogLevel.ERROR, "Error occurred creating the elementinfo object", ex);
             }
-
             return elementInfo;
         }
         private object GetElementConfigFromPayload(PayLoad payLoad)
         {
             ElementActionCongifuration configArgs = new ElementActionCongifuration();
-            //
-            configArgs.LocateBy = eLocateBy.ByXPath;
-            configArgs.LocateValue = payLoad.GetValueString();
 
-            if (!payLoad.Name.Equals("SwitchWindow"))
+            ElementInfo elementInfo;
+
+            if (payLoad.Name.StartsWith("html", StringComparison.OrdinalIgnoreCase))
```

Comment:
Compare the full payload name with equals like we are doing in all places. Because if someone will add another payload which is starting with html but it is for different use case then it will fail